stackit/internal/services/ske/cluster/resource.go

```diff
@@ -269,6 +269,17 @@ func (r *clusterResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 			"kubernetes_version": schema.StringAttribute{
 				Description: "Kubernetes version. Must only contain major and minor version (e.g. 1.22)",
 				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplaceIf(stringplanmodifier.RequiresReplaceIfFunc(func(ctx context.Context, sr planmodifier.StringRequest, rrifr *stringplanmodifier.RequiresReplaceIfFuncResponse) {
+						if sr.StateValue.IsNull() || sr.PlanValue.IsNull() {
+							return
+						}
+						planVersion := fmt.Sprintf("v%s", sr.PlanValue.ValueString())
+						stateVersion := fmt.Sprintf("v%s", sr.StateValue.ValueString())
+
+						rrifr.RequiresReplace = semver.Compare(planVersion, stateVersion) < 0
+					}), "Kubernetes version", "If the Kubernetes version is a downgrade, the cluster will be replaced"),
+				},
```

Comment:
I was wondering if it's worth implementing this is a new package like we did for the `validate` package (for validators, e.g. `validate.NoSeparator()`) but I think we can keep it here for now and then extract when we need to re-use it

Reply:
It would be a cleaner approach, but I'm not sure the overhead is worth the payoff here. Plan modifiers should be rare, especially compared to input validation.

Reply:
Yes I agree, if we later decide we want to this for more services we would extract it